### PR TITLE
tests: arch: x86: move tests to new ztest API

### PR DESCRIPTION
**Pull request by enjiamai**
Move following tests to the new ztest API:

- test/arch/common/

- tests/arch/x86/cpu_scrubs_regs

- tests/arch/x86/nmi

- tests/arch/x86/pagetables

- tests/arch/x86/static_idt

Except: test/arch/x86/info because it is not using ztest framework.

Signed-off-by: Enjia Mai <enjia.mai@intel.com>

### CODE REVIEW

File: tests/arch/x86/cpu_scrubs_regs/src/main.c [hunk 1]
```diff
@@ -69,7 +69,7 @@ static inline void z_vrfy_test_cpu_write_reg(void)
  *
  * @ingroup kernel_memprotect_tests
  */
-void test_syscall_cpu_scrubs_regs(void)
+ZTEST_USER(test_x86_cpu_scrubs_regs, test_syscall_cpu_scrubs_regs)
 {
 #if CONFIG_X86
```
Comment (yperess) [inline tests/arch/x86/cpu_scrubs_regs/src/main.c:74]: Note, `testcase.yaml` sets `arch_allow: x86` so this `#if CONFIG_X86` will always be true, can be removed

File: tests/arch/x86/pagetables/src/main.c [hunk 2]
```diff
@@ -233,7 +233,11 @@ void z_vrfy_dump_my_ptables(void)
  *
  * @ingroup kernel_memprotect_tests
  */
-void test_dump_ptables(void)
+#ifdef CONFIG_USERSPACE
+ZTEST_USER(x86_pagetables, test_dump_ptables)
+#else
+ZTEST(x86_pagetables, test_dump_ptables)
+#endif
```
Comment (yperess) [inline tests/arch/x86/pagetables/src/main.c:240]: `ZTEST_USER` will default to `ZTEST` if `CONFIG_USERSPACE=n`, you don't need this, can just use `ZTEST_USER`

**Review by yperess — CHANGES_REQUESTED** [review]

File: tests/arch/x86/pagetables/src/main.c (hunk 2 above)
Comment (enjiamai) [reply tests/arch/x86/pagetables/src/main.c]: Thanks for correcting this. Fixed.
May I consult a question with you? If I want to run test_dump_ptables() twice, one is on userspace and another is not,  is there any way I can do it? Thanks! According to the original tests:
```
        ztest_unit_test(test_dump_ptables),
        ztest_user_unit_test(test_dump_ptables)
```

File: tests/arch/x86/cpu_scrubs_regs/src/main.c (hunk 1 above)
Comment (enjiamai) [reply tests/arch/x86/cpu_scrubs_regs/src/main.c]: Thanks for pointing this out, fixed, thanks!

Comment (enjiamai) [reply tests/arch/x86/cpu_scrubs_regs/src/main.c]: Got it, I was also considering should I keep test_ or not. Seems like leave it only for testcase is better.

**Review by dcpleung — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: tests/arch/x86/pagetables/src/main.c (hunk 2 above)
Comment (yperess) [reply tests/arch/x86/pagetables/src/main.c]: Yes, you could do something like:

```
static void test_dump_ptables_common(void)
{
  // Test logic here
}

ZTEST_USER(x86_pagetables, test_dump_ptables_user)
{
  test_dump_ptables_common();
}

ZTEST(x86_pagetables, test_dump_ptables)
{
  test_dump_ptables_common();
}
```

**Review by yperess — CHANGES_REQUESTED** [review]
Waiting for the fix to run `test_dump_ptables` in both user and non-user modes.

File: tests/arch/x86/pagetables/src/main.c (hunk 2 above)
Comment (yperess) [reply tests/arch/x86/pagetables/src/main.c]: We should have this resolved before merging, as it stands this test will only run once in usermode unlike the original logic.

Comment (enjiamai) [reply tests/arch/x86/pagetables/src/main.c]: Fixed. Thanks for your example!

**Comment by enjiamai** [comment]
> Waiting for the fix to run `test_dump_ptables` in both user and non-user modes.

Got it fixed, thank for the example! The result is as the same as the old one, thanks.

**Review by nashif — DISMISSED** [review]

**Comment by nashif** [comment]
a commit per test would also be nicer.

**Comment by enjiamai** [comment]
> a commit per test would also be nicer.

ok, got it, spilt them now.

**Review by yperess — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**